File: 2026-1/bk/sesion-de-clase-04/java/seguro-vehicular/pe/edu/segurosoft/seguros/SeguroBronce.java
package pe.edu.segurosoft.seguros;

import pe.edu.segurosoft.modelo.Vehiculo;
import pe.edu.segurosoft.modelo.Marca;

public class SeguroBronce extends SeguroBasico {

    public SeguroBronce(Vehiculo vehiculo) {
        super(vehiculo);
    }

    @Override
    public double CalcularCosto() {
        double costo = 100.00;

        if (this.vehiculo.getMarca() == Marca.Toyota) {
            costo += 100.00;
        }

        return super.CalcularCosto() + costo;
    }

    @Override
    public String Descripcion() {
        return super.Descripcion() + " + cobertura contra robos";
    }
}

File: 2026-1/sesion-de-clase-05/dotnet/seguroslib/Seguros/SeguroBronce.cs
namespace Seguros;

using Modelo;

public class SeguroBronce : SeguroBasico
{
    public SeguroBronce(Vehiculo vehiculo) : base(vehiculo)
    {
    }

    public override double CalcularCosto()
    {
        double costo = 100.00;

        if (this.vehiculo.Marca == Marca.Toyota)
        {
            costo += 100.00;
        }

        return base.CalcularCosto() + costo;
    }

    public override string Descripcion()
    {
        return base.Descripcion() + " + cobertura contra robos";
    }
}
